worker/docker_image_manager.py

```diff
@@ -0,0 +1,140 @@
+import threading
+import os
+import time
+import traceback
+import logging
+import json
+
+import file_util
+from formatting import size_str
+
+logger = logging.getLogger(__name__)
+
+
+class DockerImageManager(object):
+    """
+    Manages the Docker images available on the worker, and ensures that they
+    do not fill up the disk.
+
+    The last-used dates are only tracked for images on which
+    `DockerImageManager.touch_image()` has been called. All other images will
+    be left untouched by the cleanup thread.
+
+    DependencyManager should be instantiated before DockerImageManager, to
+    ensure that the work directory already exists.
+    """
+    STATE_FILENAME = 'images-state.json'
+
+    def __init__(self, docker, work_dir, min_free_bytes=1e9,
```

Comment:
I'd give more than 1GB...some images can be more than 1GB.  I'd say 10GB, and maybe allow this flag to be passed in from the config file.  Perhaps there should be one parameter specified in the config file with a default of 10GB that is passed to both auto bundle cleanup and docker image cleanup.  So we just have to reason about how much free space we want to maintain on disk for the worker.  (Although docker and bundles could in principle live on different disks, but really they should be in the same place.)